File: Top 100 Codes/04-Important Codes related to Arrays/11-Finding the Longest Palindrome in an Array/FindingtheLongestPalindromeinanArray.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

// Function to check if a number is a palindrome
bool isPalindrome(int num) {
    int original = num, reversed = 0;
    while (num > 0) {
        reversed = reversed * 10 + num % 10;
        num /= 10;
    }
    return original == reversed;
}

// Function to find the longest palindrome in an array
int findLongestPalindrome(vector<int>& arr) {
    int longest = -1;
    for (int i = 0; i < arr.size(); i++) {
        if (isPalindrome(arr[i])) {
            longest = max(longest, arr[i]);
        }
    }
    return longest;
}

int main() {
    vector<int> arr = {121, 232, 4554, 12321, 67876, 9876};
    int longestPalindrome = findLongestPalindrome(arr);
    
    if (longestPalindrome != -1)
        cout << "The longest palindrome in the array is: " << longestPalindrome << endl;
    else
        cout << "No palindrome found in the array." << endl;
    
    return 0;
}

File: Top 100 Codes/04-Important Codes related to Arrays/11-Finding the Longest Palindrome in an Array/FindingtheLongestPalindromeinanArray.py
def is_palindrome(num):
    return str(num) == str(num)[::-1]

def find_longest_palindrome(arr):
    longest = -1
    for i in range(len(arr)):
        if is_palindrome(arr[i]):
            longest = max(longest, arr[i])
    return longest

arr = [121, 232, 4554, 12321, 67876, 9876]
longest_palindrome = find_longest_palindrome(arr)

if longest_palindrome != -1:
    print(f"The longest palindrome in the array is: {longest_palindrome}")
else:
    print("No palindrome found in the array.")
